src/main/kotlin/io/sip3/salto/ce/sdp/SdpHandler.kt

```diff
@@ -0,0 +1,175 @@
+import org.restcomm.media.sdp.SessionDescription
+import org.restcomm.media.sdp.SessionDescriptionParser
+import org.restcomm.media.sdp.attributes.RtpMapAttribute
+import org.restcomm.media.sdp.fields.MediaDescriptionField
+
+class SdpHandler : CoroutineVerticle() {
+
+    private val logger = KotlinLogging.logger {}
+
+    private var codecs = mapOf<String, Codec>()
+    private val sessions = mutableMapOf<String, SdpSessionDescription>()
+
+    override suspend fun start() {
+        readCodecs(config)
+        vertx.eventBus().localConsumer<JsonObject>(RoutesCE.config_change) { event ->
+            try {
+                val config = event.body()
+                readCodecs(config)
+            } catch (e: Exception) {
+                logger.error("SdpHandler `readCodecs()` failed.", e)
+            }
+        }
+
+        vertx.eventBus().localConsumer<SIPMessage>(RoutesCE.sdp_session_update) { event ->
+            try {
+                val message = event.body()
+                handleSdp(message)
+            } catch (e: Exception) {
+                logger.error("SdpHandler 'handleSdp()' failed.", e)
+            }
+        }
+
+        vertx.eventBus().localConsumer<String>(RoutesCE.sdp_session_terminate) { event ->
+            try {
+                val sessionKey = event.body()
+                sessions.remove(sessionKey)
+            } catch (e: Exception) {
+                logger.error("SdpHandler sdpSession remove failed.", e)
+            }
+        }
+    }
+
+    private fun readCodecs(config: JsonObject) {
+        val tmpCodecs = mutableMapOf<String, Codec>()
+        config.getJsonArray("codecs")?.forEach { codecObject ->
+            val codec = (codecObject as JsonObject).mapTo(Codec::class.java)
+            tmpCodecs[codec.name!!] = codec
+        }
+        codecs = tmpCodecs
+    }
+
+    private fun handleSdp(message: SIPMessage) {
+        val sessionDescription: SessionDescription = getSessionDescription(message) ?: return
+
+        val callId = message.callId()!!
+        sessions.getOrPut(callId) { SdpSessionDescription(callId) }.apply {
+            when (message) {
+                is SIPRequest -> {
+                    request = sessionDescription
+                }
+                is SIPResponse -> {
+                    response = sessionDescription
+                }
+            }
+
+            if (isFilled()) {
+                sendSdpSession(this)
+            }
+        }
+    }
+
+    private fun sendSdpSession(sdpSessionDescription: SdpSessionDescription) {
+        val requestMediaDescription = sdpSessionDescription.request.getMediaDescription("audio")
+        val responseMediaDescription = sdpSessionDescription.response.getMediaDescription("audio")
+
+        val callId = sdpSessionDescription.callId
+        val codec = getCodec(requestMediaDescription, responseMediaDescription)
+        val requestSdpSession = createSdpSession(callId, requestMediaDescription, codec)
```

Comment:
I don't like domain logic mixed in separate method.
Personally I would prefer to define a method `sendSdpSession` and create and send domain object inside it